### ISSUE TITLE
UnusedImports: imported types inside Javadoc generic method-reference arguments are reported as unused

### ISSUE THREAD

**Issue by gianmarcoschifone** [checkstyle/checkstyle#20547]
Follow-up to https://github.com/checkstyle/checkstyle/pull/20477.

Example:

```java
import java.math.BigDecimal;
import java.util.List;

/**
 * {@link #method(List<BigDecimal>)}
 */
public class Test {

    public void method(java.util.List<java.math.BigDecimal> value) {
    }
}
```

With `UnusedImports` and `processJavadoc=true`, Checkstyle reports:

```text
Unused import - java.math.BigDecimal.
```

Expected: no violation, because `BigDecimal` is referenced from the Javadoc link.

The problem is that generic method-reference parameter text like `List<BigDecimal>` is reduced to only `List`, so imported types used inside generic arguments are not marked as referenced.

Similar cases:

```java
{@link #method(Class<? extends BigDecimal>)}
{@link #method(Class<? super BigDecimal>)}
```

**Comment by Carbon14-48** [comment]
hello @gianmarcoschifone  can i work on this ?

**Comment by gianmarcoschifone** [comment]
> hello [@gianmarcoschifone](https://github.com/gianmarcoschifone) can i work on this ?

hi @Carbon14-48 , yes sure
